ui/components/MesheryRegistrationLogs.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, Provider } from 'react-redux';
+import { selectEvents } from '../store/slices/events';
+import { CircularProgress, Box } from '@material-ui/core';
+import axios from 'axios';
+import { store } from '../store';
+
+const Loading = () => {
+  return (
+    <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+      <CircularProgress />
+    </Box>
+  );
+};
+
+const EventsView = () => {
+  const events = useSelector(selectEvents) || [];
```

Comment:
this is  a paginated filtered view of events .  directly make a rtk api call to events endpoint with filters applied to get the log events at take the latest

Reply:
We don't have direct end-point for the logs via event. Do we need to create a separate rtk-query?

After hitting : `/api/v2/events`
Getting below:
`{"events":[{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.677091308+05:00","description":"For registrant successfully imported 9 models 520 relationships","id":"d25da155-f379-4ae1-a3b7-b8139efafc68","metadata":{"Hostname":""},"operation_id":"bc78cf3a-4ded-487d-a6c5-ac47ac57a5e3","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.677186219+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.677716596+05:00","description":"For registrant Artifacthub successfully imported 603 models 27605 components","id":"3f143fd1-b5d9-43fe-9bea-7e93f8e92ac0","metadata":{"Hostname":"Artifacthub"},"operation_id":"258fc289-69a6-43b3-a69e-2281b1779bc2","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.677797557+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"registration","created_at":"2024-07-12T21:15:33.677980533+05:00","description":"Failed to import 18 components","id":"9fe6d3d4-6312-45bc-a46a-d79a15ebe91c","metadata":{"DownloadLink":"/home/moin/.meshery/logs/registry/registry-logs.log","LongDescription":"The import process for a registrant Artifacthub encountered difficulties,due to which Failed to import 18 components. Specific issues during the import process resulted in certain entities not being successfully registered in the table.","SuggestedRemediation":"Visit docs with the error code https://docs.meshery.io/reference/error-codes","ViewLink":"/home/moin/.meshery/logs/registry/registry-logs.log"},"operation_id":"e29d9175-8800-46b9-9b09-7e108ac32880","severity":"error","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.678019699+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.678178134+05:00","description":"For registrant Github successfully imported 50 models 6005 components","id":"fa57ab57-566f-49a2-aebc-bf37a4ce4923","metadata":{"Hostname":"Github"},"operation_id":"7895c3c7-d9d8-4aaa-9b2a-31e2b80432ac","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.678224375+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"}],"page":0,"page_size":25,"count_by_severity_level":[],"total_count":4}`

Should I fetch 2 times? One for reaching events via `/api/v2/events` and getting the required url (attached with ViewLink), then hitting the target url i.e. /home/moin/.meshery/logs/registry/registry-logs.log to get the logs data?

Are you talking about directly hitting the below api?
https://playground.meshery.io/api/system/fileView?file=%2Fhome%2Fappuser%2F.meshery%2Flogs%2Fregistry%2Fregistry-logs.lo

Pretty confused, kindly elaborate a bit.

Reply:
@aabidsofi19